package-dev/Runtime/SentryIntegrations.cs

```diff
@@ -0,0 +1,142 @@
+                // TODO: Add a place to put a custom 'SceneManagerAPI' on the editor window so we can "decorate" it.
+                _logger?.Log(SentryLevel.Warning,
+                    "Registering SceneManagerTracing integration - overwriting the previous SceneManagerAPI.overrideAPI.");
+            }
+
+            SceneManagerAPI.overrideAPI = new SceneManagerTracingAPI(_logger);
+            ShouldCaptureStartup = true;
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.SubsystemRegistration)]
+        public static void SubsystemRegistration()
+        {
+            // Sentry inits here
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSplashScreen)]
+        public static void BeforeSplashScreen()
+        {
+            if (!ShouldCaptureStartup)
+            {
+                return;
+            }
+
+            StartupTransaction = SentrySdk.StartTransaction("transaction-from-startup", "Splash Screen");
+            SentrySdk.ConfigureScope(scope => scope.Transaction = StartupTransaction);
+
+            SplashAndAssembliesSpan = SentrySdk.GetSpan()?.StartChild("startup.splashscreen");
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterAssembliesLoaded)]
+        public static void AfterAssembliesLoaded()
+        {
+            // There is not really anything here for use to measure/capture?
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+        public static void BeforeSceneLoad()
+        {
+            if (!ShouldCaptureStartup)
+            {
+                return;
+            }
+
+            SplashAndAssembliesSpan.Finish(SpanStatus.Ok);
+            SceneLoadSpan = SentrySdk.GetSpan()?.StartChild("startup.scene.load");
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterSceneLoad)]
+        public static void AfterSceneLoad()
+        {
+            if (!ShouldCaptureStartup)
+            {
+                return;
+            }
+
+            SceneLoadSpan.Finish(SpanStatus.Ok);
+            StartupTransaction.Finish(SpanStatus.Ok);
+        }
+    }
+
+    public class SceneManagerTracingAPI : SceneManagerAPI
+    {
+        private readonly IDiagnosticLogger _logger;
+
+        public SceneManagerTracingAPI(IDiagnosticLogger logger)
+        {
+            _logger = logger;
+        }
+
+        protected override AsyncOperation LoadSceneAsyncByNameOrIndex(string sceneName, int sceneBuildIndex, LoadSceneParameters parameters, bool mustCompleteNextFrame)
+        {
+            _logger.LogInfo("Starting scene load transaction for '{0}'.", sceneName);
+
+            var transaction = SentrySdk.StartTransaction("transaction-from-scene-load", sceneName ?? $"buildIndex:{sceneBuildIndex}");
+            SentrySdk.ConfigureScope(scope => scope.Transaction = transaction);
+
+            var span = SentrySdk.GetSpan()?.StartChild("scene.load");
```

Comment:
No need to use null propagation if you have the transaction instance:
```suggestion
            var span = transaction.StartChild("scene.load");
```